lib/DocumentFilter/DocumentFilterFieldArray.test.js

```diff
@@ -1,53 +1,87 @@
 import { MemoryRouter } from 'react-router-dom';
+import { waitFor } from '@folio/jest-config-stripes/testing-library/react';
 
-import { renderWithIntl, Button, TestForm } from '@folio/stripes-erm-testing';
+import { renderWithIntl, Button, TestForm, IconButton } from '@folio/stripes-erm-testing';
 
 import { translationsProperties } from '../../test/jest/helpers';
+import { categoryValues } from '../../test/jest/resources';
 import DocumentFilterFieldArray from './DocumentFilterFieldArray';
 
 const onSubmit = jest.fn();
+jest.mock('./DocumentFilterField', () => () => <div>DocumentFilterField</div>);
 
-const categoryValues = [
-  {
-    'id': '2c9180a09262108601926219be050022',
-    'value': 'consortium_negotiation_document',
-    'label': 'Consortium negotiation document',
-  },
-  {
-    'id': '2c9180a09262108601926219bdfc0020',
-    'value': 'license',
-    'label': 'License',
-  },
-  {
-    'id': '2c9180a09262108601926219be010021',
-    'value': 'misc',
-    'label': 'Misc',
-  },
-];
-
+let renderComponent;
 describe('DocumentFilterFieldArray', () => {
   beforeEach(() => {
-    renderWithIntl(
+    renderComponent = renderWithIntl(
       <MemoryRouter>
         <TestForm onSubmit={onSubmit}>
           <DocumentFilterFieldArray
             categoryValues={categoryValues}
             filterName="docs"
           />
-        </TestForm>
-        ,
+        </TestForm>,
       </MemoryRouter>,
       translationsProperties
     );
   });
 
-  test('renders the document filters button', async () => {
+  test('adding/removing filterd fields using the add/remove works as expected', async () => {
+    const { getByText, queryAllByTestId } = renderComponent;
     await Button('Add filter').exists();
+    await waitFor(async () => {
+      await Button('Add filter').click();
+    });
+
+    await waitFor(async () => {
+      expect(getByText('DocumentFilterField')).toBeInTheDocument();
+    });
+
+    await waitFor(async () => {
+      expect(queryAllByTestId(/document-filter-field\[.*\]/i).length).toEqual(1);
+    });
+
+    await waitFor(async () => {
```

Comment:
Same here, this is a sequence of describes/tests, not one big lump.

Jest doesn't handle asynchronicity that well across differing environments, so try to keep a test to a single thing, and use describes etc to set up the state that we're testing.

Again let me know if you want an insight into what I mean on a call etc